includes/Abilities/DiscoverAbilitiesAbility.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * Ability for discovering available WordPress abilities.
+ *
+ * @package McpAdapter
+ */
+
+declare( strict_types=1 );
+
+namespace WP\MCP\Abilities;
+
+/**
+ * Discover Abilities - Lists all available WordPress abilities in the system.
+ *
+ * This ability provides discovery functionality for the MCP protocol.
+ * It discovers all registered WordPress abilities in the system.
+ *
+ * SECURITY CONSIDERATIONS:
+ * - This ability exposes information about all registered abilities in the system
+ * - Only abilities with mcp.public=true metadata will be returned
+ * - Requires proper WordPress capability checks for secure operation
+ *
+ * @see https://github.com/your-repo/mcp-adapter/docs/security.md for detailed security configuration
+ */
+final class DiscoverAbilitiesAbility {
+	use McpAbilityHelperTrait;
+
+	/**
+	 * Register the ability.
+	 */
+	public static function register(): void {
+		wp_register_ability(
+			'mcp-adapter/discover-abilities',
+			array(
+				'label'               => 'Discover Abilities',
+				'description'         => 'Discover all available WordPress abilities in the system. Returns a list of all registered abilities with their basic information.',
+				'category'            => 'mcp-adapter',
+				'input_schema'        => array(
+					'type'                 => 'object',
+					'additionalProperties' => false,
+				),
+				'output_schema'       => array(
+					'type'       => 'object',
+					'properties' => array(
+						'abilities' => array(
+							'type'  => 'array',
+							'items' => array(
+								'type'       => 'object',
+								'properties' => array(
+									'name'        => array( 'type' => 'string' ),
+									'label'       => array( 'type' => 'string' ),
+									'description' => array( 'type' => 'string' ),
+								),
```

Comment:
I don't necessarily think we need to if it's working, but something we can explore doing more of in things like properties is adding `description` and `examples` to the individual properties' schema to better inform the AI.